tracing-journald/src/lib.rs

```diff
@@ -155,6 +157,31 @@ impl Subscriber {
         &self.syslog_identifier
     }
 
+    /// Sets the syslog facility for this logger.
+    ///
+    /// The syslog facility comes from the classic syslog interface (`openlog()`
+    /// and `syslog()`). In syslog the facility argument is used to specify
+    /// what type of program is logging the message. This lets the configuration
+    /// file specify that messages from different facilities will be handled
+    /// differently. Systemd exposes it in the `SYSLOG_FACILITY` journal field,
+    /// and allows filtering log messages by syslog facility with `journalctl
+    /// --facility`.
+    ///
+    /// See [Journal Fields](https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html)
+    /// and [journalctl](https://www.freedesktop.org/software/systemd/man/journalctl.html)
+    /// for more information.
+    ///
+    /// If not set this field is left out.
+    pub fn with_syslog_facility(mut self, facility: SyslogFacility) -> Self {
+        self.syslog_facility = Some(facility);
```

Comment:
if I understand the documentation correctly, some syslog facilities should never be used by user programs (`SyslogFacility::Kern` and `SyslogFacility::Syslogd`). should we be checking that the provided syslog facility is not one of those, and emitting an error if it is?

alternatively, we could also just remove those variants from the enum so that users cannot construct loggers that use disallowed facilities...

Reply:
I removed the `SyslogFacility` enum as requested in another comment.